pipeline/outputs/s3.md

```diff
@@ -1,141 +1,206 @@
+- You must use `store_dir_limit_size` to limit the space on disk used by S3 buffer files.
+- The original ordering of data inputted to Fluent Bit may not be preserved unless you enable `preserve_data_ordering On`.
 
 ## S3 Key Format and Tag Delimiters
 
-In Fluent Bit, all logs have an associated tag. The `s3_key_format` option lets you inject the tag into the s3 key using the following syntax:
+In Fluent Bit, all logs have an associated tag. The `s3_key_format` option lets you
+inject the tag into the S3 key using the following syntax:
 
-* `$TAG` => the full tag
-* `$TAG[n]` => the nth part of the tag (index starting at zero). This syntax is copied from the rewrite tag filter. By default, “parts” of the tag are separated with dots, but you can change this with `s3_key_format_tag_delimiters`.
+- `$TAG`: The full tag.
+- `$TAG[n]`: The nth part of the tag (index starting at zero). This syntax is copied
+  from the rewrite tag filter. By default, “parts” of the tag are separated with
+  dots, but you can change this with `s3_key_format_tag_delimiters`.
 
-In the example below, assume the date is January 1st, 2020 00:00:00 and the tag associated with the logs in question is `my_app_name-logs.prod`.
+In the following example, assume the date is January 1st, 2020 00:00:00 and the tag
+associated with the logs in question is `my_app_name-logs.prod`.
 
-```
+```python
 [OUTPUT]
-    Name                         s3
-    Match                        *
+    Name  s3
+    Match *
     bucket                       my-bucket
     region                       us-west-2
     total_file_size              250M
     s3_key_format                /$TAG[2]/$TAG[0]/%Y/%m/%d/%H/%M/%S/$UUID.gz
     s3_key_format_tag_delimiters .-
 ```
 
-With the delimiters as . and -, the tag will be split into parts as follows:
+With the delimiters as `.` and `-,` the tag splits into parts as follows:
 
-* `$TAG[0]` = my\_app\_name
-* `$TAG[1]` = logs
-* `$TAG[2]` = prod
+- `$TAG[0]` = `my_app_name`
+- `$TAG[1]` = `logs`
+- `$TAG[2]` = `prod`
 
-So the key in S3 will be `/prod/my_app_name/2020/01/01/00/00/00/bgdHN1NM.gz`.
+The key in S3 will be `/prod/my_app_name/2020/01/01/00/00/00/bgdHN1NM.gz`.
 
 ### Allowing a file extension in the S3 Key Format with $UUID
 
-The Fluent Bit S3 output was designed to ensure that previous uploads will never be over-written by a subsequent upload. Consequently, the `s3_key_format` supports time formatters, `$UUID`, and `$INDEX`. `$INDEX` is special because it is saved in the `store_dir`; if you restart Fluent Bit with the same disk, then it can continue incrementing the index from its last value in the previous run.
+The Fluent Bit S3 output was designed to ensure that previous uploads will never be
+over-written by a subsequent upload. The `s3_key_format` supports time formatters,
+`$UUID`, and `$INDEX`. `$INDEX` is special because it is saved in the `store_dir`; if
+you restart Fluent Bit with the same disk, then it can continue incrementing the
```

Comment:
```suggestion
`$UUID`, and `$INDEX`. `$INDEX` is special because it is saved in the `store_dir`. If
you restart Fluent Bit with the same disk, it can continue incrementing the
```